### PR TITLE
[Backport release-24.11] zulip: build package from source

### PR DESCRIPTION
**Pull request by nbdd0121**
This is the manual backport of #279545, automatic backport of which failed due to #358620.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JohnRTitor** [comment]
@ofborg build zulip

**Review by JohnRTitor — COMMENTED** [review]
I forgot to mention this on the other PR, but this can likely build on aarch64-linux as well. Could you try changing the meta.platforms and check if it builds on aarch64-linux? @nbdd0121

**Comment by nbdd0121** [comment]
It builds fine, although probably that change should be a separate PR to master and then backported?

**Comment by JohnRTitor** [comment]
Yes, that's what I was asking for. Let's hold off on merging this one, just open another PR targeting master with the meta.platforms change. After that gets merged, cherry-pick the commit here.

**Review by JohnRTitor — COMMENTED** [review]
Cherry-picks should be done with the -x flag.

**Review by JohnRTitor — APPROVED** [review]

**Review by JohnRTitor — COMMENTED** [review]
Please cherry-pick 2160e7db923d9ded9787975891de5dfd07192961 instead, it is recommended to cherry-pick exactly the same changes, else issues may arise during future auto backports. I would suggest you leave the wayland-ime and env change as it is.

https://github.com/NixOS/nixpkgs/actions/runs/12203255621/job/34045797277?pr=362394

File: pkgs/by-name/zu/zulip/package.nix
Comment (andersk) [inline pkgs/by-name/zu/zulip/package.nix:23]:
```suggestion
  env = {
    ELECTRON_SKIP_BINARY_DOWNLOAD = 1;
  };
```

No reason to introduce skew vs. the cherry-picked commit here.

**Review by andersk — CHANGES_REQUESTED** [review]

**Pull request merged**
